crates/holochain_sqlite/src/db/p2p_state.rs

```diff
@@ -76,35 +60,69 @@ impl AsP2pStateConExt for crate::db::PConn {
     ) -> DatabaseResult<Vec<KitsuneAgent>> {
         self.with_reader(move |reader| reader.p2p_gossip_query(since_ms, until_ms, within_arc))
     }
+}
 
-    fn p2p_prune(&mut self) -> DatabaseResult<()> {
-        self.with_commit(move |writer| writer.p2p_prune())
-    }
+/// Put an AgentInfoSigned record into the p2p_store
+pub async fn p2p_put(db: &DbWrite, signed: &AgentInfoSigned) -> DatabaseResult<()> {
```

Comment:
Are these not in the trait to avoid dealing with BoxFutures? -- I wish they'd land async traits... it's obnoxious writing in an unfinished language...

Reply:
Yeh I didn't want to bring in async_trait. I forgot you cold just return the future from the trait. Is there any reason it needs to be a trait though?

Reply:
I just prefer methods to standalone functions as an organizational strategy to help me understand code relationships.